Src/Test.UnitTests.Sarif.PatternMatcher.Cli/AnalyzeCommandTests.cs

```diff
@@ -453,6 +548,124 @@ private SarifLog RunAnalyzeCommand(string definitionsText, string fileContents)
+            mockFileSystem.Setup(x => x.DirectoryEnumerateFiles(rootDirectory,
+                                                                scanTargetName,
+                                                                It.IsAny<SearchOption>()))
+                                                                    .Returns(new[] { scanTargetPath });
+
+            // Search definitions location and loading
+            mockFileSystem.Setup(x => x.FileExists(searchDefinitionsPath)).Returns(true);
+            mockFileSystem.Setup(x => x.FileReadAllText(It.IsAny<string>()))
+                .Returns<string>((path) =>
+                {
+                    return path == scanTargetPath ?
+                      fileContents :
+                      definitionsText;
+                });
+
+            // Shared strings location and loading
+            mockFileSystem.Setup(x => x.FileReadAllLines(It.IsAny<string>()))
+                .Returns<string>((path) => { return GetSharedStrings(); });
+
+            mockFileSystem.Setup(x => x.FileWriteAllText(It.IsAny<string>(), It.IsAny<string>()))
+                .Callback(new Action<string, string>((path, logText) => { sarifOutput = logText; }));
+
+            mockFileSystem.Setup(x => x.FileInfoLength(SmallTargetName)).Returns(fileContents.Length);
+            mockFileSystem.Setup(x => x.FileExists(@$"c:\Test.UnitTests.Sarif.PatternMatcher.Cli.dll")).Returns(true);
+            mockFileSystem.Setup(x => x.AssemblyLoadFrom(It.IsAny<string>())).Returns(Assembly.LoadFrom(dllLocation));
+
+            Program.FileSystem = mockFileSystem.Object;
+
+            string tempFileName = Path.GetTempFileName();
+            string sarifLogFileName = $"{tempFileName}.sarif";
+            SarifLog sarifLog = null;
+            string levels = "Error;Warning;Note";
+
+            try
+            {
+                string[] args = new[]
+                {
+                    "analyze",
+                    scanTargetPath,
+                    $"-d", searchDefinitionsPath,
+                    $"-o", sarifLogFileName,
+                    "--level", levels,
+                    "--rich-return-code",
+                };
+
+                if (runDynamicValidation)
+                {
+                    args.Append("--dynamic-validation");
+                }
+
+                int result = Program.Main(args);
+                result.Should().Be(0);
+
+                sarifLog = JsonConvert.DeserializeObject<SarifLog>(File.ReadAllText(sarifLogFileName));
+            }
+            finally
+            {
+                if (File.Exists(tempFileName))
+                {
+                    File.Delete(tempFileName);
+                }
+
+                if (File.Exists(sarifLogFileName))
+                {
+                    File.Delete(sarifLogFileName);
+                }
+            }
+
+            return sarifLog;
+        }
+
+        private void ValidateFailureLevelByValidationType(
```

Comment:
>ValidateFailureLevelByValidationType [](http://example.com/codeflow?start=21&length=36)

Split out because this validation was used more than once and the potential for use in other tests. Again, this is part of an effort to demonstrate one option in standardizing our test assertions.